src/app/calculator/steam/stack-loss/stack-loss.component.ts

```diff
@@ -152,6 +153,7 @@ export class StackLossComponent implements OnInit {
       }
     }
     if (this.inModal == true) {
+      this.cd.detectChanges();
```

Comment:
If you haven't already please confirm that the root cause of whatever problem requires cd.detectChanges() can't be solved another way.